sdk/Trace/RandomIdGenerator.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Sdk\Trace;
+
+class RandomIdGenerator implements IdGenerator
+{
+    private const TRACE_ID_HEX_LENGTH = 32;
+    private const SPAN_ID_HEX_LENGTH = 16;
+
+    public function generateTraceId(): string
+    {
+        return $this->randomHex(self::TRACE_ID_HEX_LENGTH);
+    }
+
+    public function generateSpanId(): string
+    {
+        return $this->randomHex(self::SPAN_ID_HEX_LENGTH);
+    }
+
+    private function randomHex(int $hexLength): string
+    {
+        try {
+            return bin2hex(random_bytes(intdiv($hexLength, 2)));
+        } catch (\Throwable $e) {
+            return $this->fallbackAlgorithm($hexLength);
+        }
+    }
+
+    private function fallbackAlgorithm(int $hexLength): string
+    {
+        return substr(str_shuffle(str_repeat('0123456789abcdef', $hexLength)), 1, $hexLength);
```

Comment:
I think this is fine for now, but we may want to consider another option here in the future.  I think it'd be a very strange use case where `bin2hex` doesn't return a correct value.

Reply:
The fallback method guards a rare case when `random_bytes()` may throw an error (https://www.php.net/manual/en/function.random-bytes.php).
The implementations of this method was copied from the [existing code](https://github.com/open-telemetry/opentelemetry-php/blob/96d0feb2d23dc9c42aef9587f4260a9e174a05bb/sdk/Trace/SpanContext.php#L186).

Reply:
Good enough for me!